examples/dev_flow/sanitizers/compiler_sanitizers.rst

```diff
@@ -0,0 +1,356 @@
+
+   include(default)
+
+   [settings]
+   build_type=Debug
+   compiler.sanitizer=AddressUndefinedBehavior
+
+   [conf]
+   tools.build:cflags+=["-fsanitize=address,undefined", "-fno-omit-frame-pointer"]
+   tools.build:cxxflags+=["-fsanitize=address,undefined", "-fno-omit-frame-pointer"]
+   tools.build:exelinkflags+=["-fsanitize=address,undefined"]
+   tools.build:sharedlinkflags+=["-fsanitize=address,undefined"]
+
+It is supported by GCC and Clang. MSVC does not support UBSan.
+
+**Source code:**
+
+.. code-block:: cpp
+   :caption: signed_integer_overflow/main.cpp
+   :emphasize-lines: 14
+
+   #include <iostream>
+   #include <cstdlib>
+   #include <climits>
+
+   int main() {
+   #ifdef __SANITIZE_ADDRESS__
+     std::cout << "Address sanitizer enabled\n";
+   #else
+     std::cout << "Address sanitizer not enabled\n";
+   #endif
+
+     int x = INT_MAX;
+     x += 42;                     // signed integer overflow
+
+     return EXIT_SUCCESS;
+   }
+
+**Build and run:**
+
+.. code-block:: bash
+
+   conan export signed_integer_overflow/
+   conan install --requires=signed_integer_overflow/0.1.0 -pr profiles/asan_ubsan -of signed_integer_overflow/install --build=missing
+   source signed_integer_overflow/install/conanrun.sh
+   signed_integer_overflow
+
+**Expected output (abbreviated):**
+
+.. code-block:: text
+
+   Address sanitizer enabled
+   .../main.cpp:16:9: runtime error: signed integer overflow: 2147483647 + 1 cannot be represented in type 'int'
+
+Passing the information to the compiler or build system
+-------------------------------------------------------
+
+Besides using Conan profiles to manage sanitizer settings, you can also use other approaches.
+
+Managing sanitizers with a CMake toolchain
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you already have a :ref:`custom CMake toolchain file <conan_cmake_user_toolchain>` to manage compiler
+and build options, you can pass the necessary flags to enable sanitizers there instead of profiles.
+
+.. code-block:: cmake
+   :caption: cmake/my_toolchain.cmake
+
+   # Apply to all targets; consider per-target options for finer control
+   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address,undefined -fno-omit-frame-pointer")
+   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address,undefined -fno-omit-frame-pointer")
+   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address,undefined")
+   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address,undefined")
+
+Then, specify this toolchain file as part of your Conan profile:
+
+.. code-block:: ini
+   :caption: profiles/asan_ubsan
+
+   include(default)
+
+   [settings]
+   build_type=Debug
+   compiler.sanitizer=AddressUndefinedBehavior
+
+   [conf]
+   tools.cmake.cmaketoolchain:user_toolchain=cmake/my_toolchain.cmake
+
+This way, you can keep your existing CMake toolchain file and still leverage Conan profiles to manage other settings.
+
+Managing sanitizers with Conan hooks
```

Comment:
I personally don't like this way of adding the sanitizers "in the shadows". I think it's much better to do this explicitly in profiles, so I think I would not document this.

Reply:
Agreed. This is documented in Conan 1.x as well, but indeed keeping it tracked in the profiles only is much safer.